scripts/exgdas_atmos_gldas.sh

```diff
@@ -0,0 +1,337 @@
+else
+ echo "JCAP=${JCAP} not supported, exit"
+ export err=4
+ ${ERRSCRIPT} || exit 4
+fi
+
+echo "${JCAP}"
+echo "${gds}"
+ymdpre=$(sh "${FINDDATE}" "${gldas_symd}" d-1)
+ymdend=$(sh "${FINDDATE}" "${gldas_eymd}" d-2)
+ymd=${ymdpre}
+
+if [[ "${USE_CFP}" = "YES" ]] ; then
+  rm -f ./cfile
+  touch ./cfile
+fi
+
+while [[ "${ymd}" -le "${ymdend}" ]]; do
+  if [[ "${ymd}" -ne "${ymdpre}" ]]; then
+    if [[ "${USE_CFP}" = "YES" ]] ; then
+      echo "${COPYGB} -i3 '-g${gds}' -x ${GDAS}/cpc.${ymd}/precip.gldas.${ymd}00 ${RUNDIR}/cmap.gdas.${ymd}00" >> ./cfile
+      echo "${COPYGB} -i3 '-g${gds}' -x ${GDAS}/cpc.${ymd}/precip.gldas.${ymd}06 ${RUNDIR}/cmap.gdas.${ymd}06" >> ./cfile
+    else
+      ${COPYGB} -i3 -g"${gds}" -x "${GDAS}/cpc.${ymd}/precip.gldas.${ymd}00" "${RUNDIR}/cmap.gdas.${ymd}00"
+      ${COPYGB} -i3 -g"${gds}" -x "${GDAS}/cpc.${ymd}/precip.gldas.${ymd}06" "${RUNDIR}/cmap.gdas.${ymd}06"
+    fi
+  fi
+  if [[ "${ymd}" -ne "${ymdend}" ]]; then
+    if [[ "${USE_CFP}" = "YES" ]] ; then
+      echo "${COPYGB} -i3 '-g${gds}' -x ${GDAS}/cpc.${ymd}/precip.gldas.${ymd}12 ${RUNDIR}/cmap.gdas.${ymd}12" >> ./cfile
+      echo "${COPYGB} -i3 '-g${gds}' -x ${GDAS}/cpc.${ymd}/precip.gldas.${ymd}18 ${RUNDIR}/cmap.gdas.${ymd}18" >> ./cfile
+    else
+      ${COPYGB} -i3 -g"${gds}" -x "${GDAS}/cpc.${ymd}/precip.gldas.${ymd}12" "${RUNDIR}/cmap.gdas.${ymd}12"
+      ${COPYGB} -i3 -g"${gds}" -x "${GDAS}/cpc.${ymd}/precip.gldas.${ymd}18" "${RUNDIR}/cmap.gdas.${ymd}18"
+    fi
+  fi
+  ymd=$(sh "${FINDDATE}" "${ymd}" d+1)
+done
+
+if [[ "${USE_CFP}" = "YES" ]] ; then
+  ${APRUN_GLDAS_DATA_PROC:?} ./cfile
+fi
+
+# create configure file
+"${USHgldas}/gldas_liscrd.sh" "${gldas_sdate}" "${iau_cdate}" "${JCAP}"
+export err=$?
+${ERRSCRIPT} || exit 4
+
+
+#---------------------------------------------------------------
+### 3) Produce initials noah.rst from 6-tile gdas restart files 
+#---------------------------------------------------------------
+rm -f fort.41 fort.141 fort.11 fort.12
+
+# 3a) create gdas2gldas input file 
+
+cat >> fort.141 << EOF
+ &config
+   data_dir_input_grid="${input1}"
+   sfc_files_input_grid="${gldas_symd}.${gldas_scyc}0000.sfcanl_data.tile1.nc","${gldas_symd}.${gldas_scyc}0000.sfcanl_data.tile2.nc","${gldas_symd}.${gldas_scyc}0000.sfcanl_data.tile3.nc","${gldas_symd}.${gldas_scyc}0000.sfcanl_data.tile4.nc","${gldas_symd}.${gldas_scyc}0000.sfcanl_data.tile5.nc","${gldas_symd}.${gldas_scyc}0000.sfcanl_data.tile6.nc"
+   mosaic_file_input_grid="${CASE}_mosaic.nc"
+   orog_dir_input_grid="${topodir}"
+   orog_files_input_grid="${CASE}_oro_data.tile1.nc","${CASE}_oro_data.tile2.nc","${CASE}_oro_data.tile3.nc","${CASE}_oro_data.tile4.nc","${CASE}_oro_data.tile5.nc","${CASE}_oro_data.tile6.nc"
+   i_target=${nlon}
+   j_target=${nlat}
+   model="${model:?}"
+ /
+EOF
+cp fort.141 fort.41
+
+
+# 3b) Use gdas2gldas to generate nemsio file 
+
+export OMP_NUM_THREADS=1
+export pgm=gdas2gldas
+# shellcheck disable=SC1091
+. prep_step
```

Comment:
We should probably look at whether this actually needs to be sourced, or just run normally. And does it actually have to be called before each step? Can be later though.

Reply:
Will check in later work.